codegen/method.go

```diff
@@ -846,69 +846,45 @@ func (ms *MethodSpec) setTypeConverters(
 	h *PackageHelper,
 	downstreamMethod *MethodSpec,
 ) error {
-	// TODO(sindelar): Iterate over fields that are structs (for foo/bar examples).
-
 	// Add type checking and conversion, custom mapping
-	structType := compile.FieldGroup(funcSpec.ArgsSpec)
-	downstreamStructType := compile.FieldGroup(downstreamSpec.ArgsSpec)
-
-	typeConverter := NewTypeConverter(h)
-
-	typeConverter.append(
-		"func convertTo",
-		pascalCase(ms.Name),
-		"ClientRequest(in ", ms.RequestType, ") ", downstreamMethod.RequestType, "{")
+	structType := &compile.StructSpec{Fields: compile.FieldGroup(funcSpec.ArgsSpec)}
+	downstreamStructType := &compile.StructSpec{Fields: compile.FieldGroup(downstreamSpec.ArgsSpec)}
 
-	typeConverter.append("out := &", downstreamMethod.ShortRequestType, "{}\n")
+	typeConverter := NewTypeConverter(h, ConvertOptions{
+		FromSuffix:       "ClientRequest",
+		FromInputType:    ms.RequestType,
+		FromOutputType:   downstreamMethod.RequestType,
+		ToType:           downstreamMethod.ShortRequestType,
+		OutputMethodName: ms.Name,
+	})
 
 	err := typeConverter.GenStructConverter(structType, downstreamStructType, reqTransforms)
 	if err != nil {
 		return err
 	}
-	typeConverter.append("\nreturn out")
-	typeConverter.append("}")
 	ms.ConvertRequestGoStatements = typeConverter.GetLines()
 
 	// TODO: support non-struct return types
 	respType := funcSpec.ResultSpec.ReturnType
-	downstreamRespType := funcSpec.ResultSpec.ReturnType
+	downstreamRespType := downstreamSpec.ResultSpec.ReturnType
```

Comment:
^ this seems like a really nuanced bug. Why did this change, is there a test case for this ?

Reply:
i think the previous was a bug 🤷‍♂️